src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -186,7 +184,7 @@ private Response createSession(Command command) throws IOException {
         refreshAdditionalCommands();
         setResponseCodec(dialect.getResponseCodec());
         Response response = result.createResponse();
-        if (this.appiumClientConfig != null && this.appiumClientConfig.isDirectConnectEnabled()) {
+        if (appiumClientConfig.isDirectConnectEnabled()) {
```

Comment:
Please keep the check. It's unrelated to this PR

Reply:
I agree it's unrelated, however the field cannot be null and was showing as warning in my IDE
* `appiumClientConfig` is passed via `super`
* and then in the base class checked with `remoteServer = Require.nonNull("HTTP client configuration", config).baseUrl();`